api/pkg/apis/v1alpha1/vendors/targets-container-vendor.go

```diff
@@ -0,0 +1,163 @@
+		Version:  o.Vendor.Version,
+		Name:     "TargetContainers",
+		Producer: "Microsoft",
+	}
+}
+
+func (e *TargetContainersVendor) Init(config vendors.VendorConfig, factories []managers.IManagerFactroy, providers map[string]map[string]providers.IProvider, pubsubProvider pubsub.IPubSubProvider) error {
+	err := e.Vendor.Init(config, factories, providers, pubsubProvider)
+	if err != nil {
+		return err
+	}
+	for _, m := range e.Managers {
+		if c, ok := m.(*targetcontainers.TargetContainersManager); ok {
+			e.TargetContainersManager = c
+		}
+	}
+	if e.TargetContainersManager == nil {
+		return v1alpha2.NewCOAError(nil, "target container manager is not supplied", v1alpha2.MissingConfig)
+	}
+	return nil
+}
+
+func (o *TargetContainersVendor) GetEndpoints() []v1alpha2.Endpoint {
+	route := "targetcontainers"
+	if o.Route != "" {
+		route = o.Route
+	}
+	return []v1alpha2.Endpoint{
+		{
+			Methods:    []string{fasthttp.MethodGet, fasthttp.MethodPost, fasthttp.MethodDelete},
+			Route:      route,
+			Version:    o.Version,
+			Handler:    o.onTargetContainers,
+			Parameters: []string{"name?"},
+		},
+	}
+}
+
+func (c *TargetContainersVendor) onTargetContainers(request v1alpha2.COARequest) v1alpha2.COAResponse {
+	pCtx, span := observability.StartSpan("onTargetContainers", request.Context, &map[string]string{
+		"method": "onTargetContainers",
+	})
+	defer span.End()
+	tcLog.Infof("V (TargetContainers): onTargetContainers, method: %s, traceId: %s", request.Method, span.SpanContext().TraceID().String())
+
+	id := request.Parameters["__name"]
+	namespace, exist := request.Parameters["namespace"]
+	if !exist {
+		namespace = constants.DefaultScope
+	}
+
+	tcLog.Infof("V (TargetContainers): onTargetContainers, method: %s, traceId: %s", string(request.Method), span.SpanContext().TraceID().String())
+	switch request.Method {
+	case fasthttp.MethodGet:
+		ctx, span := observability.StartSpan("onTargetContainers-GET", pCtx, nil)
+		var err error
+		var state interface{}
+		isArray := false
+		if id == "" {
+			// Change partition back to empty to indicate ListSpec need to query all namespaces
+			if !exist {
+				namespace = ""
+			}
+			state, err = c.TargetContainersManager.ListState(ctx, namespace)
+			isArray = true
+		} else {
+			state, err = c.TargetContainersManager.GetState(ctx, id, namespace)
+		}
+		if err != nil {
+			tcLog.Errorf("V (TargetContainers): onTargetContainers failed - %s, traceId: %s", err.Error(), span.SpanContext().TraceID().String())
+			return observ_utils.CloseSpanWithCOAResponse(span, v1alpha2.COAResponse{
+				State: v1alpha2.InternalError,
+				Body:  []byte(err.Error()),
+			})
+		}
+		jData, _ := utils.FormatObject(state, isArray, request.Parameters["path"], request.Parameters["doc-type"])
+		resp := observ_utils.CloseSpanWithCOAResponse(span, v1alpha2.COAResponse{
+			State:       v1alpha2.OK,
+			Body:        jData,
+			ContentType: "application/json",
+		})
+		if request.Parameters["doc-type"] == "yaml" {
```

Comment:
Why we need this?

Reply:
Request contract for all vendors. @Haishi2016 do we need to remove this in container vendors?

Reply:
We can keep these. These are from earlier customer feedbacks where they want to consume query result in either JSON or YAML and they want to be able to do simple extractions from the query results

Reply:
Thanks @Haishi2016. Will resolve this conversation.